client/src/components/frontend/ResourceList/Filters.js

```diff
@@ -1,17 +1,62 @@
 import React, { Component, PropTypes } from 'react';
-import { Link } from 'react-router';
 import capitalize from 'lodash/capitalize';
+import get from 'lodash/get';
 
 export default class ResourceListFilters extends Component {
 
   static displayName = "ResourceList.Filters";
 
   static propTypes = {
-    kinds: PropTypes.array
+    kinds: PropTypes.array,
+    tags: PropTypes.array,
+    filterChangeHandler: PropTypes.func.isRequired
   };
 
-  constructor() {
-    super();
+  constructor(props) {
+    super(props);
+    this.state = this.initialState();
+    this.setFilters = this.setFilters.bind(this);
+    this.resetFilters = this.resetFilters.bind(this);
+  }
+
+  componentDidUpdate(prevProps, prevState) {
+    if (get(prevState, 'filter') !== get(this.state, 'filter')) {
+      this.props.filterChangeHandler(this.state.filter);
+    }
+  }
+
+  setFilters(event, label) {
+    const key = event.target.value;
+    const filter = Object.assign({}, this.state.filter);
+    const inputs = Object.assign({}, this.state.inputs);
+    if (key === "") {
```

Comment:
There's something about this method I'm not crazy about... it feels verbose, for one. Also, there's the fact that this will always update state, even if nothing has changed in the inputs or the filters. I'm not saying rewrite it, I just think it could probably be tightened up a bit.